jest.config.js

```diff
@@ -1,5 +1,27 @@
-const { compilerOptions } = require('./tsconfig.json');
+const tsConfig = require('get-tsconfig').getTsconfig();
+const compilerOptions = tsConfig.config.compilerOptions;
+const { baseUrl, paths } = compilerOptions;
+const useSwc = !tsConfig.path.includes('packages/cli');
 
+/** @type {import('@swc/types').Config} */
+const swcJestOptions = {
+	jsc: {
+		parser: {
+			syntax: 'typescript',
+			tsx: false,
+			decorators: true,
+		},
+		transform: {
+			legacyDecorator: true,
+			decoratorMetadata: true,
+		},
+		target: 'es2020',
+	},
+};
+
+// TODO: remove ts-jest and use swc in `cli` once swc supports emitting decorator metadata
+// https://github.com/swc-project/swc/issues/6824
```

Comment:
> until `@swc/core` supports emitting decorator metadata

1. `n8n-hosted-backend/packages/dashboard-backend` also has typeorm, class-validator and class-transformer, but [my swc PR for that repo](https://github.com/n8n-io/n8n-hosted-backend/pull/280) works. Based on your explanation it shouldn't work then?
2. TS recently [implemented](https://devblogs.microsoft.com/typescript/announcing-typescript-5-2-beta/#decorator-metadata) the decorator metadata proposal. I expect this means `--emitDecoratorMetadata` is no longer experimental and could be removed. Unfamiliar with the details of decorator metadata, but wondering if this is somehow relevant?

Reply:
looks like we had this working in n8n as well in [this PR](https://github.com/n8n-io/n8n/pull/6320/commits/0490f097f8fa90f5f96623d5cbe1b9034b79332f) (that I have no recollection of 🤦🏽 )